framework/Optimizers/SPSA.py

```diff
@@ -156,13 +158,14 @@ def localStillReady(self, ready, convergence = False):
         # check to see if the grad evaluation points have all been collected
         evalsFinished = True
         for pertID in range(self.gradDict['pertNeeded']):
-          if not self._checkModelFinish(traj,self.counter['varsUpdate'][traj],pertID)[0]:#[0]:
+          if not self._checkModelFinish(traj,self.counter['varsUpdate'][traj],pertID)[0]:
             evalsFinished = False
             break
         # if grad eval pts are finished, then evaluate the gradient
         if evalsFinished:
           # collect output values for perturbed points
-          for i in range(1,self.gradDict['numIterForAve']*2,2):
+          #for i in range(1,self.gradDict['numIterForAve']*2,2):
+          for i in self.perturbationIndeces:
```

Comment:
This doesn't need to be changed per se, but for future reference (at least in American English) we spell it "indices."  Again, just trivia, no change needed.

Reply:
Ops...misspelled...I will change it